launcher/container_runner.go

```diff
@@ -122,6 +123,26 @@ func NewRunner(ctx context.Context, cdClient *containerd.Client, token oauth2.To
 		return nil, err
 	}
 
+	if launchSpec.MonitoringEnabled != spec.None {
+		logger.Printf("Health Monitoring is enabled by the VM operator")
+
+		if launchSpec.MonitoringEnabled == spec.All {
+			logger.Printf("All health monitoring metrics enabled")
+			if err := nodeproblemdetector.EnableAllConfig(); err != nil {
+				logger.Printf("Failed to enable full monitoring config: %v", err)
+				return nil, err
+			}
+		} else if launchSpec.MonitoringEnabled == spec.MemoryOnly {
+			logger.Printf("memory/bytes_used enabled")
+		}
+
+		if err := nodeproblemdetector.StartService(logger); err != nil {
+			logger.Print(err)
+		}
+	} else {
+		logger.Printf("Health Monitoring is disabled")
+	}
+
```

Comment:
nit: consider extracting this chunk of logics to a new function as `NewRunner` is becoming too complicated.